gobblin-core/src/main/java/gobblin/source/extractor/extract/google/webmaster/GoogleWebmasterClientImpl.java

```diff
@@ -0,0 +1,98 @@
+import com.google.api.services.webmasters.WebmastersScopes;
+import com.google.api.services.webmasters.model.ApiDataRow;
+import com.google.api.services.webmasters.model.ApiDimensionFilter;
+import com.google.api.services.webmasters.model.ApiDimensionFilterGroup;
+import com.google.api.services.webmasters.model.SearchAnalyticsQueryRequest;
+import com.google.api.services.webmasters.model.SearchAnalyticsQueryResponse;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class GoogleWebmasterClientImpl implements GoogleWebmasterClient {
+  private final static Logger LOG = LoggerFactory.getLogger(GoogleWebmasterClientImpl.class);
+
+  private final Webmasters.Searchanalytics _analytics;
+
+  public GoogleWebmasterClientImpl(String credentialFile, String appName, String scope) throws IOException {
+    Preconditions.checkArgument(
+        Objects.equals(WebmastersScopes.WEBMASTERS_READONLY, scope) || Objects.equals(WebmastersScopes.WEBMASTERS,
+            scope), "The scope for WebMaster must either be WEBMASTERS_READONLY or WEBMASTERS");
+
+    GoogleCredential credential =
+        GoogleCredential.fromStream(new FileInputStream(credentialFile)).createScoped(Collections.singletonList(scope));
+    //GoogleNetHttpTransport.newTrustedTransport(),
+    //JacksonFactory.getDefaultInstance(),
+    Webmasters service =
+        new Webmasters.Builder(new NetHttpTransport(), new JacksonFactory(), credential).setApplicationName(appName)
+            .build();
+    _analytics = service.searchanalytics();
+  }
+
+  @Override
+  public List<String> getPages(String siteProperty, String date, GoogleWebmasterFilter.Country country, int rowLimit,
+      List<GoogleWebmasterFilter.Dimension> requestedDimensions, List<ApiDimensionFilter> filters, int startRow)
+      throws IOException {
+    checkRowLimit(rowLimit);
+    Preconditions.checkArgument(requestedDimensions.contains(GoogleWebmasterFilter.Dimension.PAGE));
+
+    SearchAnalyticsQueryResponse rspByCountry =
+        searchAnalyticsQuery(siteProperty, date, requestedDimensions, GoogleWebmasterFilter.andGroupFilters(filters),
+            rowLimit, startRow);
+
+    List<ApiDataRow> pageRows = rspByCountry.getRows();
+    List<String> pages = new ArrayList<>(rowLimit);
+    if (pageRows != null) {
+      LOG.info(String.format("%d pages fetched for market-%s on %s. The last page has %.1f clicks.", pageRows.size(),
+          country, date, pageRows.get(pageRows.size() - 1).getClicks()));
+
+      int pageIndex = requestedDimensions.indexOf(GoogleWebmasterFilter.Dimension.PAGE);
+      for (ApiDataRow row : pageRows) {
+        pages.add(row.getKeys().get(pageIndex));
+      }
+    } else {
+      LOG.info(String.format("0 pages fetched for market-%s on %s.", country, date));
```

Comment:
This LOG.info won't be called often, maybe 4~5 times per market per day. And the information is kind of valuable for now. Instead, I've removed logging in doQuery() at GoogleWebmasterDataFetcherImpl. They are too trivial to keep.